keras_applications/unet.py

```diff
@@ -0,0 +1,206 @@
+    https://arxiv.org/abs/1505.04597)
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+from . import get_keras_submodule
+
+backend = get_keras_submodule('backend')
+engine = get_keras_submodule('engine')
+layers = get_keras_submodule('layers')
+models = get_keras_submodule('models')
+
+
+def double_conv3x3(input_tensor,
+                   filters,
+                   activation,
+                   padding,
+                   block_name,
+                   conv_index=1):
+    """ Apply successively two convolution layers
+
+    # Arguments
+        input_tensor: input tensor
+        filters: integer, number of filters on convolution layers
+        activation: activation function name on convolution layers
+        padding: padding to use on convolution layers
+        block: current block name
+        conv_index: current index of the convolution,
+            to build the right name for layer
+
+    # Returns
+        Output tensor for the block
+    """
+    x = layers.Conv2D(filters, (3, 3),
+                      activation=activation,
+                      padding=padding,
+                      name=block_name + '_conv' + str(conv_index)
+                      )(input_tensor)
+    conv_index += 1
+    x = layers.Conv2D(filters, (3, 3),
+                      activation=activation,
+                      padding=padding,
+                      name=block_name + '_conv' + str(conv_index))(x)
+
+    return x
+
+
+def double_conv3x3_maxpool(input_tensor, filters, activation, padding, block_name):
+    """ Apply successively two convolution layers and a max-pooling
+
+    # Arguments
+        input_tensor: input tensor
+        filters: integer, number of filters on convolution layers
+        activation: activation function name on convolution layers
+        padding: padding to use on convolution layers
+        block: current block name
+
+    # Returns
+        Output tensor for the block
+    """
+    conv = double_conv3x3(input_tensor, filters, activation, padding, block_name)
+    x = layers.MaxPooling2D((2, 2),
+                            strides=(2, 2),
+                            name=block_name + '_pool')(conv)
+
+    return x, conv
+
+
+def UNet(input_tensor=None,
+         input_shape=None,
+         padding='valid',
+         weights=None):
+    """ Instanciates the UNet architecture.
+
+    Optionally loads weights pre-trained on Biomedical Image segmentation dataset ?
+
+    # Arguments
+        input_tensor: optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+        input_shape: optional shape tuple, only to be specified
+            if `include_top` is False (otherwise the input shape
+            has to be `(224, 224, 3)` (with `channels_last` data format)
+            or `(3, 224, 224)` (with `channels_first` data format).
+            It should have exactly 3 inputs channels,
```

Comment:
Where is `exactly 3 input channels` required? The original method would work on any number of channels.

Reply:
Sorry, I copied the docstring from other architectures and didn't spot this. Indeed, the architecture can take less than 3 channels, I'll impact this soon